jinahub/rankers/DPRReaderRanker/dpr_reader.py

```diff
@@ -0,0 +1,231 @@
+                scores = {'relevance_score': rel_score}
+                new_match = Document(text=span, scores=scores, tags=tags)
+                new_matches.append(new_match)
+                new_match_ref_doc_dict[new_match.id] = ref_ind
+
+        # Clear old matches
+        for doc in docs_list:
+            doc.matches.clear()
+
+        # Append new matches, first sorting them
+        for ref_ind, new_matches_group in groupby(
+            new_matches, lambda x: new_match_ref_doc_dict[x.id]
+        ):
+            new_matches_list = list(new_matches_group)
+            new_matches_list.sort(
+                key=lambda x: (x.scores['relevance_score'].value, x.tags['span_score']),
+                reverse=True,
+            )
+
+            docs_list[ref_ind].matches.extend(new_matches_list)
+
+    def _prepare_inputs(
+        self,
+        matches: List[Document],
+        docs_list: Dict[str, Document],
+        old_match_ref_doc_dict: Dict[str, str],
+    ) -> Tuple[List[str], List[str], Optional[List[str]], List[str]]:
+        contexts = []
+        questions = []
+        ref_inds = []
+        for match in matches:
+            contexts.append(match.text)
+            ref_inds.append(old_match_ref_doc_dict[match.id])
+            questions.append(docs_list[old_match_ref_doc_dict[match.id]].text)
+
+        titles = None
+        if self.title_tag_key:
+            try:
+                titles = [match.tags[self.title_tag_key] for match in matches]
+            except KeyError:
+                raise ValueError(
+                    f'All matches are required to have the {self.title_tag_key} tag,'
+                    ' but found somem atches without it.'
+                )
+
+        return questions, contexts, titles, ref_inds
+
+    def _get_outputs(
+        self,
+        questions: List[str],
+        contexts: List[str],
+        titles: Optional[List[str]],
+        ref_inds: List[str],
+    ) -> Tuple[List[float], List[str], List[Dict[str, Union[str, float]]], List[str]]:
+
+        encoded_inputs = self.tokenizer(
+            questions=questions,
+            titles=titles,
+            texts=contexts,
+            padding='longest',
+            return_tensors='pt',
+        )
+        outputs = self.model(**encoded_inputs)
+
+        # For each context, extract num_spans_per_match best spans
+        best_spans = self.tokenizer.decode_best_spans(
+            encoded_inputs,
+            outputs,
+            num_spans=self.num_spans_per_match * len(questions),
+            num_spans_per_passage=self.num_spans_per_match,
+        )
+        relevance_scores = _logistic_fn([span.relevance_score for span in best_spans])
+        spans = [span.text for span in best_spans]
+        answer_ref_inds = [ref_inds[span.doc_id] for span in best_spans]
+
+        tags = []
+        for span in best_spans:
+            tags_span = {'span_score': _logistic_fn(span.span_score)}
```

Comment:
Instead of storing `span_score`, I think putting it into `scores['span_score']` is more intuitive.

Reply:
Alright